Wikipedia/UI-V5/WMFSearchViewController.m

```diff
@@ -99,59 +116,85 @@ - (void)setRecentSearchesHidden:(BOOL)hidingRecentSearches animated:(BOOL)animat
         self.resultsListContainerView.alpha = 1.0 - self.recentSearchesContainerView.alpha;
-    }];
+    } completion:nil];
 }
 
 - (void)configureArticleList {
-    NSParameterAssert(self.dataStore);
-    NSParameterAssert(self.recentPages);
-    NSParameterAssert(self.savedPages);
     self.resultsListController.dataStore   = self.dataStore;
-    self.resultsListController.recentPages = self.recentPages;
-    self.resultsListController.savedPages  = self.savedPages;
+    self.resultsListController.recentPages = self.dataStore.userDataStore.historyList;
+    self.resultsListController.savedPages  = self.dataStore.userDataStore.savedPageList;
+    self.resultsListController.delegate    = self;
 }
 
 - (void)configureRecentSearchList {
-    NSParameterAssert(self.recentSearches);
-    self.recentSearchesViewController.recentSearches = self.recentSearches;
+    self.recentSearchesViewController.recentSearches = self.dataStore.userDataStore.recentSearchList;
     self.recentSearchesViewController.delegate       = self;
 }
 
 #pragma mark - UIViewController
 
+- (BOOL)prefersStatusBarHidden {
+    return NO;
+}
+
+- (void)configureSearchField {
+    [self.closeButton wmf_setButtonType:WMFButtonTypeX];
+    [self setSeparatorViewHidden:YES animated:NO];
+    // TODO: localize
+    [self.searchField setPlaceholder:@"Search Wikipedia"];
+}
+
 - (void)viewDidLoad {
     [super viewDidLoad];
 
+    [self configureSearchField];
+
+    self.searchFieldTop.constant = -self.searchFieldHeight.constant;
+    [self.view layoutIfNeeded];
+
     // TODO: localize
     self.title                                                    = @"Search";
     self.resultsListController.collectionView.keyboardDismissMode = UIScrollViewKeyboardDismissModeOnDrag;
     SelfSizingWaterfallCollectionViewLayout* resultLayout = [self.resultsListController flowLayout];
     resultLayout.minimumLineSpacing = 5.f;
     [self updateUIWithResults:nil];
-
-    // TODO: localize
-    [self.searchBar setPlaceholder:@"Search Wikipedia"];
-    self.searchBar.tintColor       = [UIColor darkGrayColor];
-    self.searchBar.searchBarStyle  = UISearchBarStyleMinimal;
-    self.searchBar.backgroundColor = [UIColor colorWithRed:0.9294 green:0.9294 blue:0.9294 alpha:1.0];
 }
 
 - (void)viewWillAppear:(BOOL)animated {
     [super viewWillAppear:animated];
-    [self updateRecentSearchesVisibility:animated];
+
+    [self.searchFieldTop setConstant:0];
+
+    BOOL willAnimate = [self.transitionCoordinator
+                        animateAlongsideTransition:^(id < UIViewControllerTransitionCoordinatorContext > _Nonnull context) {
+        [self.view layoutIfNeeded];
+        [self updateRecentSearchesVisibility:animated];
+        [self.searchField becomeFirstResponder];
+        [self setSeparatorViewHidden:NO animated:YES];
+    }
+                                        completion:nil];
+    NSParameterAssert(willAnimate);
```

Comment:
i mostly added it because this is the first time i've done this inside a view controller's `viewWillAppear:`. but, if this is the place to do this kind of things and it works, I think the assertion is unnecessary on the grounds that you wouldn't assert the result of this method when called inside something like `viewWillTransitionToSize:`